app/models/SDESNotificationRecord.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.notification.{RecordStatus, SDESNotification}
+import play.api.libs.json.{JsNumber, JsValue, Json, OFormat, Reads, Writes}
+
+import java.time.LocalDateTime
+
+
+case class SDESNotificationRecord(reference: String,
+                                   status: RecordStatus.Value = RecordStatus.PENDING,
+                                  numberOfAttempts: Int = 0,
+                                  createdAt:LocalDateTime = LocalDateTime.now(),
+                                  updatedAt: LocalDateTime = LocalDateTime.now(),
+                                  nextAttemptAt: LocalDateTime = LocalDateTime.now(),
+                                  notification: SDESNotification)
+
+object SDESNotificationRecord {
+  implicit object DefaultJodaDateWrites extends Writes[org.joda.time.DateTime] {
+    def writes(d: org.joda.time.DateTime): JsValue = JsNumber(d.getMillis)
+  }
+
+  implicit val writes: Writes[SDESNotificationRecord] = Json.writes[SDESNotificationRecord]
+
+  implicit val reads: Reads[SDESNotificationRecord] = Json.reads[SDESNotificationRecord]
+
+  implicit val mongoFormats: OFormat[SDESNotificationRecord] = Json.format[SDESNotificationRecord]
```

Comment:
If you have an OFormat then you don't need separate Reads or Writes - Format/OFormat reads and writes